Documentation/ApiOverview/Localization/LocalizationApi/LanguageService.rst

```diff
@@ -12,5 +12,37 @@ be created directly, therefore its constructor is internal. Create a
 
 In the backend context a :php:`LanguageService` is stored in the global
 variable :php:`$GLOBALS['LANG']`.
+In the frontend a :php:`LanguageService` can be accessed via the contentObject:
+
+..  code-block:: php
+    :caption: Classes/Controller/ExampleController.php
+
+    use Psr\Http\Message\ServerRequestInterface;
+    use TYPO3\CMS\Core\Localization\LanguageServiceFactory;
+    use TYPO3\CMS\Core\Utility\GeneralUtility;
+
+    class ExampleController {
```

Comment:
```suggestion
    final class ExampleController {
```

as best practise.

Reply:
IMO that will break any try to extend extensions with `EXT:extender`. For me it's not best practice. If you are absolutely sure that no-one has and should extend that controller because of security, special data or what ever...only that case I'm with you.

Reply:
EXT:extender is about Extbase, not non-Extbase. "final" protects against unintended extending the controller. If a developer wants this later, they have to remove the final, hence should know why. So, I see this as a safe-guard. IMHO controllers should not be extended by others (maybe EXT:extender may be an exception).

Reply:
See discussion below